src/P3_processes.jl

```diff
@@ -64,53 +64,46 @@ end
 Returns the melting rate of ice (QIMLT in Morrison and Mildbrandt (2015)).
 """
 function ice_melt(
-    p3::PSP3,
+    dist::P3Distribution{FT},
     Chen2022::CMP.Chen2022VelType,
     aps::CMP.AirProperties{FT},
     tps::TDP.ThermodynamicsParameters{FT},
-    L_ice::FT,
-    N_ice::FT,
     Tₐ::FT,
     ρₐ::FT,
-    F_rim::FT,
-    ρ_rim::FT,
     dt::FT,
 ) where {FT}
-    # process not dependent on F_liq
+    # Note: process not dependent on `F_liq`
     # (we want ice core shape params)
-    F_liq_ = FT(0)
     # Get constants
     (; ν_air, D_vapor, K_therm) = aps
     L_f = TD.latent_heat_fusion(tps, Tₐ)
     N_sc = ν_air / D_vapor
 
-    # Get the P3 diameter distribution...
-    th = thresholds(p3, ρ_rim, F_rim)
-    (λ, N_0) =
-        distribution_parameter_solver(p3, L_ice, N_ice, ρ_rim, F_rim, F_liq_)
-    N(D) = N′ice(p3, D, λ, N_0)
-    # ... and D_max for the integral
-    bound = get_ice_bound(p3, λ, N_ice, FT(1e-6))
+    state = get_state(dist)
+    params = get_parameters(dist)
```

Comment:
Not sure if its a good idea to "repackage" the parameters into a bigger struct. Maybe it would be better to keep passing smaller structs and here pass only the ones we need.